io.c

```diff
@@ -8159,13 +8171,33 @@ rb_io_printf(int argc, const VALUE *argv, VALUE out)
 
 /*
  *  call-seq:
- *     printf(io, string [, obj ... ])    -> nil
- *     printf(string [, obj ... ])        -> nil
+ *    printf(io = $stdout, string, *objects) -> nil
```

Comment:
You need a separate call-seq line for the io case.  the call-seq you are using is not valid Ruby syntax (cannot have a required parameter between an optional parameter and a splat).

Reply:
Fixed.